src/azure-cli/azure/cli/command_modules/maps/custom.py

```diff
@@ -54,13 +74,124 @@ def list_accounts(client, resource_group_name=None):
     return client.list_by_resource_group(resource_group_name)
 
 
-def generic_update_account(instance, sku_name=None, tags=None):
-    # Pre-populate with old instance
-    maps_account_create_params = MapsAccountCreateParameters(location=ACCOUNT_LOCATION, sku=instance.sku,
-                                                             tags=instance.tags)
-    # Update fields with new parameter values
-    if sku_name:
-        maps_account_create_params.sku.name = sku_name
-    if tags:
-        maps_account_create_params.tags = tags
-    return maps_account_create_params
+def maps_account_show(client,
+                      resource_group_name,
+                      account_name):
+    return client.get(resource_group_name=resource_group_name,
+                      account_name=account_name)
+
+
+def maps_account_update(client,
+                        resource_group_name,
+                        account_name,
+                        tags=None,
+                        kind=None,
+                        disable_local_auth=None,
+                        linked_resources=None,
+                        type_=None,
+                        user_assigned_identities=None,
+                        name=None):
+    if kind is None:
+        kind = "Gen1"
+    if disable_local_auth is None:
+        disable_local_auth = False
+    maps_account_update_parameters = {}
+    maps_account_update_parameters['tags'] = tags
+    maps_account_update_parameters['kind'] = "Gen1" if kind is None else kind
+    maps_account_update_parameters['disable_local_auth'] = False if disable_local_auth is None else disable_local_auth
+    maps_account_update_parameters['linked_resources'] = linked_resources
+    maps_account_update_parameters['identity'] = {}
+    maps_account_update_parameters['identity']['type'] = type_
+    maps_account_update_parameters['identity']['user_assigned_identities'] = user_assigned_identities
+    maps_account_update_parameters['sku'] = {}
+    maps_account_update_parameters['sku']['name'] = name
+    return client.update(resource_group_name=resource_group_name,
+                         account_name=account_name,
+                         maps_account_update_parameters=maps_account_update_parameters)
```

Comment:
The update method generated by CodeGen is invalid. Please use generic update instead.